src/main/java/net/neoforged/neoforge/common/extensions/IFriendlyByteBufExtension.java

```diff
@@ -37,4 +43,57 @@ default void writeFluidStack(FluidStack stack) {
     default FluidStack readFluidStack() {
         return !self().readBoolean() ? FluidStack.EMPTY : FluidStack.readFromPacket(self());
     }
+
+    /**
+     * Writes the entries in the given set to the buffer, by first writing the count and then writing each entry.
+     *
+     * @param set    The set to write
+     * @param writer The writer to use for writing each entry
+     * @param <T>    The type of the entry
+     * @implNote This is a convenience method for {@link FriendlyByteBuf#writeCollection(Collection, FriendlyByteBuf.Writer)}, where the callback can be a method on the entry type.
+     */
+    default <T> void writeObjectCollection(Collection<T> set, BiConsumer<T, FriendlyByteBuf> writer) {
+        self().writeCollection(set, (buf, t) -> writer.accept(t, buf));
+    }
+
+    /**
+     * Reads an {@link ItemStack} from the current buffer, but allows for a larger count than the vanilla method, using a variable length int instead of a byte.
+     *
+     * @return The read stack
+     */
+    default ItemStack readItemWithLargeCount() {
```

Comment:
If we are just adding a couple general use extensions that aren't actually used by this PR, one that would be of some use to me rather than having to have it in my own util method is similar to `writeCollection` is there was a `writeArray` and `readArray` (I was initially going to make one for those two in a PR after this networking refactor, but if you are including the large count item ones, might as well do the other two simple ones as well)

Reply:
Done